umh-core/pkg/config/config.go

```
@@ -99,9 +102,39 @@ func (c NmapServiceConfig) Equal(other NmapServiceConfig) bool {
+	clone := FullConfig{
+		Agent:              c.Agent,
+		DataFlowComponents: make([]DataFlowComponentConfig, len(c.DataFlowComponents)),
+	}
+	// deep copy the location map if it exists
+	if c.Agent.Location != nil {
+		clone.Agent.Location = make(map[int]string)
+		for k, v := range c.Agent.Location {
+			clone.Agent.Location[k] = v
+		}
+	}
+	deepcopy.Copy(&clone.DataFlowComponents, &c.DataFlowComponents)
```

golangci-lint (1.64.8)

Error return value of `deepcopy.Copy` is not checked

(errcheck)